src/Paramore.Brighter/MessageMapperRegistry.cs

```diff
@@ -74,11 +74,16 @@ public MessageMapperRegistry(IAmAMessageMapperFactory? messageMapperFactory,
             if (_messageMapperFactory is null)
                 return null;
 
-            var messageMapperType = _messageMappers.ContainsKey(typeof(TRequest))
-                ? _messageMappers[typeof(TRequest)]
-                : _defaultMessageMapper;
-
-            if (messageMapperType is null) return null;
+            if (!_messageMappers.TryGetValue(typeof(TRequest), out var messageMapperType) && _defaultMessageMapper != null)
+            {
+                messageMapperType = _defaultMessageMapper.MakeGenericType(typeof(TRequest));
+                _messageMappers.Add(typeof(TRequest), messageMapperType);
+            }
```

Comment:
Now that we are updating the map in runtime, should I change it to `ConcurrentDictionary`?

Reply:
Yes please

Reply:
Done